docs/setup/aks/getting-started-aks.md

```diff
@@ -0,0 +1,149 @@
+
+2. Install [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest).
+
+
+
+## Set up your AKS cluster
+
+1.  **Create a reource group**: 
+
+    ```sh
+    az group create --name <resource-group-name> --location <location>
+    ```
+    For example:
+    ```sh
+    az group create --name my-nuclio-k8s-rg --location westeurope
+    ```
+2.  **Create Kuberenetes Cluster**: The following snippet creates a cluster named `myNuclioCluster` in a Resource Group named `my-nuclio-k8s-rg`. This Resource Group was created in the previous step. [Read more about creating cluster options here](https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az_aks_create).
+
+    ```sh
+    az aks create --resource-group <resource-group-name> --name <cluster-name> --node-count <number>
+    ```
+    For example:
+    ```sh
+    az aks create --resource-group my-nuclio-k8s-rg --name myNuclioCluster --node-count 2 --generate-ssh-keys
+    ```
+    After several minutes, the deployment completes, and returns json formatted information about the AKS deployment.
+
+3. **Install the kubectl CLI**: Optional if you don't have it already installed. To connect to the Kubernetes cluster from your client computer, use [kubectl](https://kubernetes.io/docs/user-guide/kubectl/), the Kubernetes command-line client. To install it locally, run the following command:
+```sh
+az aks install-cli
+```
+
+4. **Connect with kubectl**: To configure kubectl to connect to your Kubernetes cluster, run the following command:
+```sh
+az aks get-credentials --resource-group=<resource-group-name> --name=<cluster-name>
+```
+For example:
+```sh
+az aks get-credentials --resource-group=my-nuclio-k8s-rg --name=myNuclioCluster
+```
+
+5. **Verify connection to your cluster**: Run the kubectl get nodes command:
+following command:
+```sh
+kubectl get nodes
+```
+The expected output should be similar to:
+```sh
+NAME                             STATUS    AGE       VERSION
+k8s-myNuclioCluster-36346190-0   Ready     49m       v1.7.7
+```
+
+## Create a container registry using the Azure CLI
+[Azure Container Registry (ACR)](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli) is a managed Docker container registry service used for storing private Docker container images.
+
+The nuclio playground builds and pushes functions to a Docker registry. In our case we use ACR as out Docker registry.
+
+Create an ACR instance using the [az acr create](https://docs.microsoft.com/en-us/cli/azure/acr#az_acr_create) command.
+
+The name of the registry **must be unique**. In the following example `mynuclioacr` is used. Update this to a unique value.
+```sh
+az acr create --resource-group <resource-group-name> --name <registry-name> --sku Basic
+```
+For example:
+```sh
+az acr create --resource-group my-nuclio-k8s-rg --name mynuclioacr --sku Basic
+```
+
+## Granting Kubernetes access to ACR
```

Comment:
1. When you create AKS, a service principal is created for you automatically with password or you can enter it manually (In the portal you must enter the SP but with the CLI the SP can be generated automatically. This service principal is responsible of managing the infrastructure of AKS (VM's, Disks, etc...)

https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough-portal
https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough


I don't think this SP should be the same as the ACR SP since it serves a very different role in the user architecture. The user's application shouldn't use the AKS SP with the permissions to manage the cluster itself.

So, if you create a different SP for ACR, you should definitely do 
`az role assignment create --scope /subscriptions/.../resourceGroups/my-nuclio-k8s-rg/providers/Microsoft.ContainerRegistry/registries/mynuclioacr --role Owner --assignee <clientId>`

2. You can log in to a registry directly via individual login, and your applications or Kubernetes can login with unattended, or "headless," authentication by using SP. 
I think we should add the following link and let the users decide how they want to authenticate with ACR.
https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication

Once SP is created, you can't see the password again. You can only regenerate it.